homeassistant/helpers/device_registry.py

```diff
@@ -413,14 +417,20 @@ def to_device_entry(
     ) -> DeviceEntry:
         """Create DeviceEntry from DeletedDeviceEntry."""
         return DeviceEntry(
+            area_id=self.area_id,
```

Comment:
Where are these new attributes restored?

Reply:
They are restored in `DeletedDeviceEntry.to_device_entry`

Reply:
`self.area_id` isn't set before that call or in this method that I can see.

Reply:
Never mind, they are set from line 1264.